### PR TITLE
WP 169 consume click cookie

### PR DESCRIPTION
**Pull request by mmcgahan**
This PR updates the server to consume the click-tracking cookie generated by fetchUtils that contains all the click-tracking data in State.

### CODE REVIEW

**Comment by mmcgahan** [comment]
the bulk of the changes in the diff are from refactoring the activity-tracking-specific plugin to be more general - separating the logging from the serialization/encoding, which actually makes the whole thing easier to understand and test IMO

File: src/util/avro.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
+const avro = require('avsc');
+
+const click = {
```
Comment (willh-meetup) [inline src/util/avro.js]: Might be nice to say where we copied this schema from too?

File: src/util/serverUtils.js [hunk 2]
```diff
@@ -21,6 +23,8 @@ export function checkForDevUrl(value) {
 }
 
 export function onRequestExtension(request, reply) {
+	request.id = uuid.v4();
```
Comment (willh-meetup) [inline src/util/serverUtils.js]: Does this generate a new UUID for this?

File: src/util/tracking.js [hunk 3]
```diff
@@ -172,7 +172,7 @@ export const logTrack = platformAgent => (response, trackInfo) => {
 	const eventDate = new Date();
 	const record = {
 		timestamp: eventDate.getTime().toString(),
-		requestId: uuid.v4(),
+		requestId: response.request.id,
```
Comment (willh-meetup) [inline src/util/tracking.js]: Ah, I see... so this means that both click/activity tracking share a UUID right?

Comment (willh-meetup) [inline src/util/tracking.js]: (I'm assuming that's what we want)

File: src/util/avro.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
+const avro = require('avsc');
```
Comment (willh-meetup) [inline src/util/avro.js]: 👍 This is a nice refactor.

File: src/util/tracking.js (hunk 3 above)
Comment (mmcgahan) [inline src/util/tracking.js]: yeah, the Hapi request always had a unique ID, but it wasn't formatted like a UUID, which I think is why we were generating a unique UUID in this step previously. Now, the `onRequestExtension` will overwrite the default `request.id` with a UUID, which can be used consistently for tracking and general logging.

I did some investigation about whether it is 'okay' to override the default `request.id` and everything I saw suggested that it shouldn't be an issue.

**Pull request merged**